main.go

```diff
@@ -35,9 +35,13 @@ func main() {
 
 	// get new conch and setup cleanup
 	c := newConch(fig)
+	hasClosedChannel := false
 
 	sm.Set(func(s *sigmon.SignalMonitor) {
-		close(c.doneChan())
+		if !hasClosedChannel {
```

Comment:
My resistance to this approach is due to the adding of a non-obvious dependency to the conch type's usage.